### PR TITLE
Add dynamic bundle size calculations

### LINKED ISSUE #46: Dynamic bundle sizes
It appears that between `master` and `v2` support for showing info about the dynamic bundle chunks was removed. Was that intentional? Would you be open to receiving a PR to add that back?

### PR DESCRIPTION
**Pull request by canac**
## Context

Re-add support for dynamic bundle sizes. Most of the code was copied as-is from `master`.

Please verify that my code actually filters out dynamic bundles inside of `node_modules`. Our codebase does not have bundles in `node_modules`, so I have no good way of testing it.

Closes #46

## Checklist
- [x] Change meets or does not compromise the [Baseline Security Requirements](https://transferwise.atlassian.net/wiki/spaces/EKB/pages/434929973/Baseline+Security+Requirements)

### CODE REVIEW

File: src/comments.ts [hunk 1]
```diff
@@ -28,13 +28,15 @@ export async function createOrReplaceComment({
...
   routesTable,
+  dynamicTable,
   strategy,
 }: {
   octokit: Octokit;
   issueNumber: number;
   title: string;
   shaInfo: string;
   routesTable: string | null;
+  dynamicTable: string | null;
```
Comment (Wise-StenRaudmets) [inline src/comments.ts:39]: Can you add a null check for `dynamicTable` on line 63 as well similar to `routesTable` so a comment is added to the PR if dynamic imports have significant changes?
https://github.com/transferwise/actions-next-bundle-analyzer/blob/8b799e0d0b76dec088d509908c41440995864da1/src/comments.ts#L63

File: src/bundle-size.ts [hunk 2]
```diff
@@ -37,6 +48,33 @@ function loadBuildManifest(workingDir: string): BuildManifest {
...
+  const content = JSON.parse(file) as ReactLoadableManifest;
+  const pages = {} as BuildManifest['pages'];
+  Object.keys(content).map((item) => {
+    const fileList = getFiles(content[item]);
+    const uniqueFileList = Array.from(new Set(fileList));
+    pages[item] = uniqueFileList.filter(
+      (file) =>
+        !appChunks.find((chunkFile) => file === chunkFile && !file.includes('/node_modules/')),
+    );
+  });
```
Comment (Wise-StenRaudmets) [inline src/bundle-size.ts]: The node_modules path is actually in the keys of `react-loadable-manifest.json` so you can filter it out there. Also you can use `.forEach` if you don't care about the return value to be clearer about the intent.
```suggestion
  Object.keys(content).forEach((item) => {
    if (item.includes('/node_modules/')) {
      return;
    }
    const fileList = getFiles(content[item]);
    const uniqueFileList = Array.from(new Set(fileList));
    pages[item] = uniqueFileList.filter(
      (file) => !appChunks.find((chunkFile) => file === chunkFile),
    );
  });
```

**Comment by canac** [comment]
@Wise-StenRaudmets Thanks for the feedback!

**Review by Wise-StenRaudmets — APPROVED** [review]
Looks good to me, thank you for contributing.

**Pull request merged**